### DISCUSSION TITLE
Configuration Variables - Which Container(s)

### DISCUSSION THREAD

**Discussion by undaunt** [karakeep-app/karakeep#271] [Q&A] [answered]
Hello,

In lieu of passing an .env file to each container, I'm passing variables from my master .env file in my compose folder. To that end, I'd like to know which container(s) specific variables need to get added to. Otherwise, I'll just add all of them to all 3 that are listed in the sample compose that declare the .env (meilisearch, workers, and web) but that will be very verbose.

Thank you.

Specifically, the default/required variables (meili master key, nextauth secret, nextauth url) as well as:

disable signups, max asset size, inference text and image models, and any of the crawler configs.

I saw this: https://github.com/hoarder-app/hoarder/discussions/187 which answers the second half of my question, but I wanted to confirm on disable signups, max size, and the three primary required.

Thank you.

**Comment by MohamedBassem** [comment, marked as answer]
Looking at: https://docs.hoarder.app/configuration

- Anything under inference config and crawler config is only used by workers container.
- Redis variables are used by both web and workers.
- NextAuth variables are only used by web.
- DATA_DIR is used by both web and workers
- max assets size, disable signups: are web only.
- MEILI_ADDR is used by both web and workers.
- MEILI_MASTER_KEY is used by web, workers and meili

hope this answers your question. We probably should add them to the docs.

**Reply by undaunt** [reply]
Thank you!
